using-timescaledb/continuous-aggregates.md

```diff
@@ -18,118 +18,162 @@ re-calculate all of the data on every refresh. Only new and/or
+view that summarizes the hourly data.  updated hourly, you would
+write:
+
+```sql
+CREATE MATERIALIZED VIEW conditions_summary_hourly
 WITH (timescaledb.continuous) AS
 SELECT device,
-       time_bucket(INTERVAL '1 day', time) AS bucket,
+       time_bucket(INTERVAL '1 hour', time) AS bucket,
        AVG(temperature),
        MAX(temperature),
        MIN(temperature)
 FROM conditions
 GROUP BY device, bucket;
+
+-- Create the policy
+SELECT add_continuous_aggregate_policy('conditions_summary_hourly',
+	start_interval => INTERVAL '1 month',
+	end_interval => NULL,
+	schedule_interval => INTERVAL '5 min');
 ```
 
+A `time_bucket` on the time partitioning column of the hypertable is
+required in all continuous aggregate views. If you do not provide one,
+you will get an error.
+
+When the view is created, it will (by default) be populated with data
+so that it contain an aggregate of the current data in
+`conditions`. If you make changes to the hypertable and want to
+refresh an interval of the continuous aggregate, you can explicity
+refresh the continuous aggregate by calling
+[`refresh_continuous_aggregate`][refresh_continuous_aggregate] and
+give the range you want to refresh:
+
+```sql
+CALL refresh_continuous_aggregate('conditions_summary_daily, '2020-05-01', '2020-06-01');
+```
+
+This will update the buckets in the range that were invalidated by
+changes in the underlying hypertable, but not touch other buckets of
+the continuous aggregate, so it does not necessarily mean that all
+buckets in the range are touched. If there are just a few
+invalidations in the range, the refresh will be quick.
+
+Note that the end range is exclusive, so this will refresh the date
+range `['2020-05-01', '2020-06-01')`, that is, up to but not including
+`2020-06-01`. While it is possible to use `NULL` to indicate an
+open-ended range, we do not recommend doing so for tables that see
+continuous ingest of new data.
+
+You should avoid refreshing time intervals that still see a lot of
+writes, which is usually the last bucket of the continuous
+aggregate. These intervals are still changing and will not produce
+accurate aggregate anyway and refreshing unnecessarily will increase
+the write amplification, which will slow down the ingest rate of the
+hypertable.
+
+```sql
+CALL refresh_continuous_aggregate('conditions_summary_daily, NULL, NULL);
+```
+
+However, note that this might materialize a lot of data, can affect
+other policies such as the data retention, and affects write
+amplification, so it should only be used rarely.
+
+>:TIP: If you want to check what invalidations there are you can use the following query:
+>```sql
+> WITH
+>  hypertables AS (
+>    SELECT
+>      id AS hypertable_id,
+>      format('%1$I.%2$I', schema_name, table_name)::regclass AS hypertable
+>    FROM
+>      _timescaledb_catalog.hypertable
+>  )
+>SELECT
+>  hypertable,
+>  _timescaledb_internal.to_timestamp(lowest_modified_value) AS start_time,
```

Comment:
There are several issues with this query.

1. It only works on timestamp-based tables
2. It only shows invalidations in the hypertable invalidation log and not the invalidations in the materialization invalidation log.
3. Showing raw invalidations might show a lot of duplicates, since the logs aren't deduped until a refresh runs.

Reply:
Removed.